Update for Cocos2d 3.0

PaulSolt opened the pull request:
I have it working for Cocos2d 3.0 sample project, but I'm not exactly how to add legacy support. The legacy projects do not build with these changes. 

How do you recommend handling the API differences from 2.x to 3.x?
1. CCArray doesn't exist in 3.0 I think this can be safely used as a normal for loop on 1.x and 2.x 
   
   ```
   for(object in array)
   ```
2. userInteractionEnabled has been added to CCNode
   
   I'm not sure the best approach for adding backwards compatibility with the API change from 2.x to 3.x. I get build errors if I try to do this:
   
   ```
   if([self respondsToSelector:@selector(isRunning)]) {
        BOOL value = [self isRunning];
   }
   ```

PaulSolt committed:
Added Cocos2d 3.0 Sample Project

Added default Cocos2d 3.x project

PaulSolt committed:
Added Cocos2d 2.x Gesture Code

Added test code to the current HelloWorldScene to create objects with
gestures.

Linking to CCNode+SFGestureRecognizers from Xcode project.

Has build errors that will be fixed in next commit.

PaulSolt committed:
Fixed Build Errors with Cocos2d 3.0

Docs aren’t clear, but I think this is the correct API transition.

Changed self.isRunning to the new method self.isRunningInActiveScene

CCArray has been replaced with just a NSArray. Using normal for() loop
to iterate children.

PaulSolt committed:
Updated .gitignore

Added *.xcuserdatad

PaulSolt commented:
The two new methods are <pre>userInteractionEnabled</pre> and <pre>isRunningInActiveScene</pre> on CCNode.

Should those properties be overridden and then called from the forwardInvocation: and methodSignatureForSelector: methods?

https://github.com/krzysztofzablocki/CCNode-SFGestureRecognizers/blob/master/CCNode%2BSFGestureRecognizers/CCNode%2BSFGestureRecognizers.m#L525

krzysztofzablocki commented:
I think you could implement it by just keeping old isTouchEnabled inside all code, just modify code for forwardInvocation and methodSignature, so if selector is not found then you first check if the instance supports the 3.0 methods. If so then you use them. Does that make sense?

Something along the lines of:

```
if (![self respondsToSelector:aSelector]) {
    if (aSelector == @selector(isTouchEnabled)) {
      if([self respondsToSelector:@selector(userInteractionEnabled)]) {
            return [self methodSignatureForSelector:@selector(userInteractionEnabled)];
      }
      return [self methodSignatureForSelector:@selector(sf_isTouchEnabled)];
...
```

PaulSolt committed:
Added support for Cocos2d 1.x and 2.x

Added method calls depending on what version of Cocos2d is being used.
Checks to see if the CCNode instance respondsToSelector.

1. isRunning on Cocos2d 1.x and 2.x
2. isRunningInActiveScene and userInteractionEnabled on Cocos2d 3.x

Removed my comments from the CCArray to NSArray loop changes.
Removed my comments from sf_isPointInArea and isPointTouchableInArea.

PaulSolt commented:
Ok, it's working for Cocos2d 1.x, 2.x, and 3.x.

I didn't change the forwardInvocation and methodSignature. Instead I added calls to check if the current instance responds to a selector and changed the methods:

```
 - (BOOL)sf_isPointInArea:(CGPoint)pt
```

and 

```
 - (BOOL)isPointTouchableInArea:(CGPoint)pt
```

to call the appropriate method (isRunning, isRunningInActiveScene, and isUserInteractionEnabled).
